### PR TITLE
Add admin organization update

### PR DESCRIPTION
**Pull request by jdenquin**
## Context

- We want to use an admin tool to be able to do basic operations
- The first one to be covered is to be able to rename an organization

## Description

- Add the `/admin` api logic
- It uses the `googleauth` gem and it's based on a Google Oauth2
- Add the `PUT /admin/organizations/:id` endpoint and logic

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,7 @@ gem 'sidekiq'
 # Security
 gem 'bcrypt'
 gem 'jwt'
+gem 'googleauth', require: false
```
Comment (jdenquin) [inline Gemfile:19]: Bundler/OrderedGems: Gems should be sorted in an alphabetical order within their section of the Gemfile. Gem `googleauth` should appear before `jwt`.

File: app/services/admin/organizations/update_service.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Admin
+  module Organizations
+    class UpdateService < ::BaseService
+      def initialize(organization:, params:)
+        @organization = organization
+        @params = params
+
```
Comment (jdenquin) [inline app/services/admin/organizations/update_service.rb:9]: Layout/TrailingWhitespace: Trailing whitespace detected. (https://rubystyle.guide#no-trailing-whitespace)

File: config/environments/development.rb [hunk 3]
```diff
@@ -47,6 +47,7 @@
 
   config.hosts << 'api.lago.dev'
   config.hosts << 'api'
+  config.hosts << "lago.ngrok.dev"
```
Comment (jdenquin) [inline config/environments/development.rb:50]: Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. (https://rubystyle.guide#consistent-string-literals)

File: spec/requests/admin/base_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Admin::BaseController, type: [:request, :admin] do
```
Comment (jdenquin) [inline spec/requests/admin/base_spec.rb:5]: RSpec/FilePath: Spec path should end with `admin/base_controller*_spec.rb`. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/FilePath)

File: spec/requests/admin/organizations_spec.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Admin::OrganizationsController, type: [:request, :admin] do
```
Comment (jdenquin) [inline spec/requests/admin/organizations_spec.rb:5]: RSpec/FilePath: Spec path should end with `admin/organizations_controller*_spec.rb`. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/FilePath)

File: app/services/admin/organizations/update_service.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Admin
+  module Organizations
+    class UpdateService < ::BaseService
+      def initialize(organization:, params:)
+        @organization = organization
+        @params = params
+  
+        super(nil)
```
Comment (rsempe) [inline app/services/admin/organizations/update_service.rb:10]:
```suggestion
        super
```

File: app/services/admin/organizations/update_service.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      end
+
+      def call
```
Comment (rsempe) [inline app/services/admin/organizations/update_service.rb:13]:
```suggestion
      def call
        return result.not_found_failure!(resource: 'organization') unless organization
```

File: spec/requests/admin/organizations_spec.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  let(:organization) { create(:organization) }
+
+  describe 'update' do
```
Comment (rsempe) [inline spec/requests/admin/organizations_spec.rb:8]:
```suggestion
  describe 'PUT /admin/organizations/:id' do
```

File: spec/services/admin/organizations/update_service_spec.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    {
+      name: 'FooBar',
+    }
+  end
+
+  describe '#call' do
+    it 'updates the organization' do
+      result = update_service.call
+
+      expect(result.organization.name).to eq('FooBar')
```
Comment (rsempe) [inline spec/services/admin/organizations/update_service_spec.rb:20]: expect { update_service.call }.to change { organization.reload.name }.to('FooBar')

Comment (rsempe) [inline spec/services/admin/organizations/update_service_spec.rb:20]:
```suggestion
      expect { update_service.call }.to change { organization.reload.name }.to('FooBar')
```

Comment (rsempe) [inline spec/services/admin/organizations/update_service_spec.rb:20]:
```suggestion
      expect { update_service.call }.to change { organization.reload.name }.to('FooBar')
```

File: app/controllers/admin/organizations_controller.rb [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        json: ::V1::OrganizationSerializer.new(
+          result.organization,
+          root_name: 'organization',
+        ),
+      )
+    end
+
+    private
+
+    def current_organization
```
Comment (rsempe) [inline app/controllers/admin/organizations_controller.rb:23]:
```suggestion
    def organization
```

**Review by rsempe — APPROVED** [review]

**Pull request merged**